### DISCUSSION TITLE
What will happen if both of manual instrumentation and automatic instrumentation exist in same java process?

### DISCUSSION THREAD

**Discussion by aaron-ai** [open-telemetry/opentelemetry-java-instrumentation#6005] [Q&A] [answered]
Whether they can be remain independent of each other ?

**Comment by jkwatson** [comment]
moving this over to the instrumentation repo

**Comment by trask** [comment, marked as answer]
hi @aaron-ai! the javaagent will hijack the `GlobalOpenTelemetry` and then all of your manual instrumentation (that uses GlobalOpenTelemetry) will interop with the agent instrumentation

**Reply by aaron-ai** [reply]
Thanks for your reply!! so they can remain independent if I use `OpenTelemetrySdk.builder()` to build my own `OpenTelemetrySdk` rather than using `GlobalOpenTelemetry` directly, am I right?

**Reply by mateuszrzeszutek** [reply]
Not exactly -- the agent also hijacks `Context` static methods, e.g. `Context#current()` (`Span#current()` is affected by that too). You might still get some interference even if you use your own `OpenTelemetrySdk` instance.

You can [disable](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-auto-instrumentation) the `opentelemetry-api` and `opentelemetry-instrumentation-api` instrumentations to completely turn off the OpenTelemetry API bridging. This way, the agent won't interfere with your app's copy of OpenTelemetry API & SDK in any way.
